### PR TITLE
Support relative path in breadcrumb_path

### PR DESCRIPTION
**Pull request by qinezh**
@chenkennt @vwxyzh @superyyrrzz @ansyral

### CODE REVIEW

**Comment by qinezh** [comment]
Three ways to use breadcrumb_path:

1. `"breadcrumb_path": "/a/b/toc.json"` (absolute path of generated TOC file)
2. `"breadcrumb_path": "~/c/toc.yml" `(relative path of source TOC file)
3. `"breadcrumb_path": "c/toc.yml" ` (will get warning that it needs `~`)

File: src/Microsoft.DocAsCode.Build.Engine/Constants.cs [hunk 1]
```diff
@@ -17,5 +17,7 @@ internal static class Constants
 
         public const string ManifestFileName = "manifest.json";
 
+        public const string BreadCrumbPath = "breadcrumb_path";
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.Engine/Constants.cs]: BreadcrumbPath

File: src/Microsoft.DocAsCode.Build.Engine/TemplateModelTransformer.cs [hunk 2]
```diff
@@ -176,6 +177,32 @@ internal ManifestItem Transform(InternalManifestItem item)
             return appended;
         }
 
+        private IDictionary<string, object> ResolveBreadCrumb(IDictionary<string, object> model)
```
Comment (chenkennt) [inline src/Microsoft.DocAsCode.Build.Engine/TemplateModelTransformer.cs]: >ResolveBreadCrumb [](start = 44, length = 17)

Make this a general function for resolve "path" metadata in model.

Comment (chenkennt) [inline src/Microsoft.DocAsCode.Build.Engine/TemplateModelTransformer.cs]: And we should let template to decide which metadata should be treated as path.

---
In reply to: [72409208](https://github.com/dotnet/docfx/pull/538#discussion_r72409208) [](ancestors = 72409208)

**Pull request closed without merging**